src/main/java/com/example/ms/card/service/concrete/CardServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+package com.example.ms.card.service.concrete;
+
+import com.example.ms.card.aspect.annotation.LogDto;
+import com.example.ms.card.aspect.annotation.LogExecutionTime;
+import com.example.ms.card.cache.CacheService;
+import com.example.ms.card.client.UserClient;
+import com.example.ms.card.dao.entity.CardEntity;
+import com.example.ms.card.dao.repository.CardRepository;
+import com.example.ms.card.exception.NotFoundException;
+import com.example.ms.card.model.enums.CardStatus;
+import com.example.ms.card.model.request.CardRequestDto;
+import com.example.ms.card.model.response.CardResponseDto;
+import com.example.ms.card.model.response.GetCardsResponseDto;
+import com.example.ms.card.service.abstraction.CardService;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.example.ms.card.cache.CacheKeyPrefixConstants.MS_CARD;
+import static com.example.ms.card.cache.CacheKeyPrefixConstants.MS_CARD_USER;
+import static com.example.ms.card.exception.ExceptionConstants.CARD_NOT_FOUND_CODE;
+import static com.example.ms.card.exception.ExceptionConstants.CARD_NOT_FOUND_MESSAGE;
+import static com.example.ms.card.exception.ExceptionConstants.USER_NOT_FOUND_CODE;
+import static com.example.ms.card.exception.ExceptionConstants.USER_NOT_FOUND_MESSAGE;
+import static com.example.ms.card.mapper.CardMapper.CARD_MAPPER;
+import static com.example.ms.card.model.enums.CardStatus.ACTIVE;
+import static com.example.ms.card.model.enums.CardStatus.BLOCKED;
+import static com.example.ms.card.model.enums.CardStatus.DELETED;
+import static com.example.ms.card.model.enums.CardStatus.STOLEN;
+
+@Service
+@RequiredArgsConstructor
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+public class CardServiceImpl implements CardService {
+    CardRepository cardRepository;
+    UserClient userClient;
+    CacheService cacheService;
+
+    @LogDto
+    @Override
+    public CardResponseDto createCard(Long userId, CardRequestDto requestDto){
+        if(!userClient.userExists(userId)){
+            throw new NotFoundException(USER_NOT_FOUND_CODE, String.format(USER_NOT_FOUND_MESSAGE, userId));
+        }
+
+        var newCardEntity = CARD_MAPPER.buildCardEntity(userId, requestDto);
+        newCardEntity = cardRepository.save(newCardEntity);
+        cacheService.saveCardToCache(newCardEntity);
+        cacheService.saveUserCardToCache(userId, newCardEntity);
+        return CARD_MAPPER.toCardResponseDto(newCardEntity);
+    }
+
+    @LogDto
+    @Override
+    public CardResponseDto getCardById(Long id){
+        CardEntity cardEntity;
+        if(cacheService.getFromCache(MS_CARD + id) != null){
```

Comment:
burda 2 defe sorgu gedir cache-a bunu 1-e endir + cache-in cokse umumi cokeceksen hec bir try catch yoxdu. Yaxwi olardiki service layer-da nese bununla bagli fikirlewesen